pkg/cmd/workflow.go

```diff
@@ -263,13 +263,23 @@ func doDownloadWorkflow(cmd *cobra.Command, args []string) error {
 	body, err := Client.DownloadWorkflow(name)
 
 	if err != nil {
-		// TODO: This error should be translated for the user. Right now it just
-		// says whatever the service says.
 		return err
 	}
 
-	// Just plop the whole thing out output.
-	Dialog.WriteString(body)
+	filepath, nil := cmd.Flags().GetString("file")
+
+	if filepath == "" {
+		// Just plop the whole thing out output.
+		Dialog.WriteString(body)
+	} else {
+		// make a byte array out of the body to feed to the writer
+		contents := []byte(body)
+		err := ioutil.WriteFile(filepath, contents, 0644)
+		if err != nil {
+			debug.Logf("failed to write to file %s: %s", filepath, err.Error())
+			return err
+		}
+	}
```

Comment:
The commentary here is a bit worthless :) We should remove both of them. You can also make it a bit more terse.

```suggestion
	if filepath == "" {
		Dialog.WriteString(body)
	} else {
		if err := ioutil.WriteFile(filepath, []byte(body), 0644); err != nil {
			debug.Logf("failed to write to file %s: %s", filepath, err.Error())
			return err
		}
	}
```